worker/src/test/scala/org/apache/celeborn/service/deploy/MiniClusterFeature.scala

```diff
@@ -98,22 +139,65 @@ trait MiniClusterFeature extends Logging {
     }
   }
 
-  def setUpMiniCluster(
+  private def setUpMiniCluster(
       masterConf: Map[String, String] = null,
       workerConf: Map[String, String] = null,
       workerNum: Int = 3): (Master, collection.Set[Worker]) = {
     val master = createMaster(masterConf)
-    val masterThread = runnerWrap(master.rpcEnv.awaitTermination())
+    val masterStartedSignal = Array(false)
+    val masterThread = new RunnerWrap({
+      try {
+        masterStartedSignal(0) = true
+        master.rpcEnv.awaitTermination()
+      } catch {
+        case ex: Exception =>
+          masterStartedSignal(0) = false
+          throw ex
+      }
+    })
     masterThread.start()
     masterInfo = (master, masterThread)
-    Thread.sleep(5000L)
-    (1 to workerNum).foreach { _ =>
-      val worker = createWorker(workerConf)
-      val workerThread = runnerWrap(worker.initialize())
-      workerThread.start()
-      workerInfos.put(worker, workerThread)
+    Thread.sleep(20000L)
```

Comment:
This can make it slow, maybe use a loop to check every 5s and break if `masterStartedSignal` is true

Reply:
sure, I changed the logic, it first wait for 5000, if master still not restarted, wait for at most 20 more seconds